lib/ukreloc/arch/x86_64/include/arch/reloc.h

```diff
@@ -54,6 +54,159 @@
 #endif
 .endm
 
+#define SAVE_REG32_COUNT				6
+
+/* Expects the lower 32 bits of the base virtual address in %edi
+ * and the higher 32 bt in %esi (hopefully a KASLR randomized value).
+ * Also place the base load address in %edx. The macro can either create its
+ * own scratch stack to save register state or use the already existing one,
+ * if there.
+ *
+ * @param have_stack Boolean value to tell whether the caller already has a
+ *                   stack available or not
+ */
+.macro do_ukreloc32	have_stack:req
```

Comment:
I just noticed that this is going to break if, in the future, we make use of the MEMRT_KERNEL memory regions related to the ELF Segments as, unlike the C written relocator, it does not relocate the ELF Segment memory regions...

Unsure whether it is worth doing this iteration now or later when the time we will actually make use of the ELF Segment memory regions comes.

Reply:
Introduced [`do_ukreloc_kmrds`](https://github.com/unikraft/unikraft/pull/772/commits/16ac1dbffc94f7570aab07c4f858893e6bf4cc54) for this.